### DISCUSSION TITLE
MQTT 5 Support

### DISCUSSION THREAD

**Discussion by gdsrmygdsrjr** [bertmelis/espMqttClient#129] [General]
Some news about when will there be support for MQTT 5?

**Comment by bertmelis** [comment]
I once started working on MQTT5 but it is on the bottom of the to do list. So sorry, no ETA whatsoever.
I find it very complicated for a protocol that is aimed at constrained devices.

Progress can be tracked here: https://github.com/bertmelis/espMqttClient/tree/mqtt5
(or you're welcome to use the branch as a starting point for further development)

**Reply by proddy** [reply]
I read somewhere that Home Assistant will drop support for MQTT 3.x in January 2027. Is it worth looking at MQTT5 again if we bring some people in to help?

**Reply by bertmelis** [reply]
wow. that would be huge. Brokers can support multiple MQTT versions but if HASS relies on new features, things might break.

I had the idea of adding MQTT5 support but then I didn't have a use case myself so stopped working on it.
